### DISCUSSION TITLE
"Hello Tokio" step of tutorial and "into" method

### DISCUSSION THREAD

**Discussion by mvolkmann** [tokio-rs/tokio#3525] [Q&A] [answered]
In this line from the "Hello Tokio" step, where is the `into` method implemented that presumably creates a `Bytes` value from a `&str` value?

```rust
    client.set("hello", "world".into()).await?;
```

**Comment by barkanido** [comment, marked as answer]
the `set` fn is of type 
```rust
pub async fn set(&mut self, key: &str, value: Bytes) -> crate::Result<()>
```
which means that the `&str` "world" is converted into `Bytes` using `into()`. `into` is automatically generated by the compiler because the `Bytes` struct is implementing several `from` methods. see [here](https://docs.rs/bytes/1.0.1/bytes/struct.Bytes.html#impl-From%3C%26%27static%20%5Bu8%5D%3E).

This is the idiomatic way for type conversions in Rust. For more information about how it works, you can read about the [From](https://doc.rust-lang.org/std/convert/trait.From.html) trait.
